### PR TITLE
Add permission checker constants and logic to hide navigation elements

### PR DESCRIPTION
**Pull request by karelhala**
### Features
Applications wants to hide navigation based on certain rules, we have to help them in this by providing helper functions and also hide these nav items if they are acutally not accessible.
### Notes
* [x] Add tests
* There is todo about api caching, but that can be ignored for now as this introduces mainly logic for checking if something can be shown.
* There is edge case when all second level navigation can be hidden, we'd probably place About in such places. But this About is not yet designed and it is not part of this PR.
### JIRA
RHCLOUD-2629

### CODE REVIEW

**Comment by karelhala** [comment]
Sorry guys for such a big change, but I had to rewrite bunch of functions to use async/await because it was super complicated or it used array push in places where it was not necessary.

File: src/js/consts.js [hunk 1]
```diff
@@ -1,7 +1,56 @@
...
+    isEntitled: async (appName) => {
+        const { entitlements } = await insights.chrome.auth.getUser();
+        return appName ? Boolean(entitlements[appName] && entitlements[appName].is_entitled) :
+            // eslint-disable-next-line camelcase
+            Object.entries(entitlements).reduce((acc, [key, { is_entitled }]) => ({ ...acc, [key]: is_entitled }), {});
+    },
+    isProd: () => {
+        return insights.chrome.isProd;
+    },
+    isBeta: () => {
```
Comment (rvsia) [inline src/js/consts.js]: What about just `() => insights.chrome.isBeta()`? same above

File: src/js/consts.js [hunk 2]
```diff
@@ -1,7 +1,56 @@
...
+        return insights.chrome.isBeta();
+    },
+    apiRequest: async ({ url, method, ...props }) => {
```
Comment (Hyperkid123) [inline src/js/consts.js]: options

File: src/js/nav/globalNav.js [hunk 3]
```diff
@@ -1,47 +1,63 @@
...
+    return [];
 }
 
 // Gets the app's data from the master config, if it exists
-function getAppData(appId, propName, masterConfig) {
+async function getAppData(appId, propName, masterConfig) {
     const app = masterConfig[appId];
     if (app && Object.prototype.hasOwnProperty.call(app, 'frontend')) {
-        const routes = getRoutesForApp(app, masterConfig);
+        if (app.permissions && !app.permissions.apps) {
```
Comment (PanSpagetka) [inline src/js/nav/globalNav.js]: Is `!app.permissions.apps` this check necessary?

Comment (karelhala) [reply src/js/nav/globalNav.js]: Yes, this hides the entire section. So if dev does not specify apps it is ment to hide the entire firt level navigation.

File: src/js/consts.test.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
+/* eslint-disable camelcase */
```
Comment (Hyperkid123) [inline src/js/consts.test.js]: Maybe we could disable this rule globally
